build_caddy.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+mkdir caddy || true
+cd caddy || true
+curl "https://raw.githubusercontent.com/caddyserver/caddy/v2/cmd/caddy/main.go" > main.go
+sed -i.bak 's/^)/	_ "github.com\/caddyserver\/nginx-adapter"\'$'\n)/' main.go && rm main.go.bak
+
+go mod init caddy
+grep 'require' ../go.mod >> go.mod
```

Comment:
Not sure I follow this one

Reply:
The go command keeps requiring beta9, so I'm just copying the `require` line to depend on the same version the adapter depends on. This go.mod file is created on-demand, so we can't fix it to a certain version. If you remember, we had similar issue in my initial PR.

Reply:
`go mod edit` might be a better, more proper way to do that, maybe? https://golang.org/cmd/go/#hdr-Edit_go_mod_from_tools_or_scripts

Or even `go get` if all we're doing is setting "require"

Reply:
Ok, used `go mod edit`. I used `go list` to read out the currently required version of Caddy so we don't need to change it in many places on upgrades. Looks better?